benchmark_aux.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2016, Linaro Limited
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <libgen.h>
+#include <linux/limits.h>
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "benchmark_aux.h"
+
+/* Misc auxilary functions */
+void tee_errx(const char *msg, TEEC_Result res)
+{
+	fprintf(stderr, "%s: 0x%08x\n", msg, res);
+	exit(1);
+}
+
+void tee_check_res(TEEC_Result res, const char *errmsg)
+{
+	if (res != TEEC_SUCCESS)
+		tee_errx(errmsg, res);
+}
+
+const char *bench_str_src(uint64_t source)
```

Comment:
This wasn't a good suggestion. Since the defined values or not simply from 0 to 4, you cannot use a lookup table. Having that said is there a good reason to have the values defined as:
```c
+/* OP-TEE susbsystems ids */
 +#define TEE_BENCH_CLIENT	0x10000000
 +#define TEE_BENCH_KMOD		0x20000000
 +#define TEE_BENCH_CORE		0x30000000
 +#define TEE_BENCH_UTEE		0x40000000
 +#define TEE_BENCH_DUMB_TA	0xF0000001
```